### PR TITLE
Zippie solidity repo

### PR DESCRIPTION
**Pull request by p4ker**

### CODE REVIEW

File: test/Token/ERC20/ZippieInternalTokenERC20.test.js [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+    it("allows whitelistAdmins to add whitelisted accounts", async function () {
+      expect(await this.token.isWhitelistAdmin(owner)).to.be.equal(true);
+      expect(await this.token.isWhitelisted(sender)).to.be.equal(false);
+      const { logs } = await this.token.addWhitelisted(sender, { from: owner });
+      expectEvent.inLogs(logs, "WhitelistedAdded", { account: sender });
+      expect(await this.token.isWhitelisted(sender)).to.be.equal(true);
+    });
+
+    it("reverts id a non-whitelistAdmins tries to add whitelisted accounts", async function () {
```
Comment (ripern) [inline test/Token/ERC20/ZippieInternalTokenERC20.test.js]: id -> if

File: test/Token/ERC20/ZippieInternalTokenERC20.test.js [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    it("allows whitelistAdmins to remove whitelisted accounts", async function () {
+      expect(await this.token.isWhitelistAdmin(owner)).to.be.equal(true);
+      await this.token.addWhitelisted(sender, { from: owner });
+      expect(await this.token.isWhitelisted(sender)).to.be.equal(true);
+      const { logs } = await this.token.removeWhitelisted(sender, { from: owner });
+      expectEvent.inLogs(logs, "WhitelistedRemoved", { account: sender });
+      expect(await this.token.isWhitelisted(sender)).to.be.equal(false);
+    });
+
+    it("reverts id a non-whitelistAdmins tries to remove whitelisted accounts", async function () {
```
Comment (ripern) [inline test/Token/ERC20/ZippieInternalTokenERC20.test.js]: id -> if

File: test/Token/ERC20/ZippieInternalTokenERC20.test.js [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+      
+        it("allows minters to renounce", async function() {
+          expect(await this.token.isMinter(owner)).to.be.equal(true);
+          const { logs } = await this.token.renounceMinter( { from: owner} );
+          expectEvent.inLogs(logs, "MinterRemoved", { account: owner });
+          expect(await this.token.isMinter(owner)).to.be.equal(false);
+        });
+      })
+    
+      describe('ERC20 Brunable', function() {
```
Comment (ripern) [inline test/Token/ERC20/ZippieInternalTokenERC20.test.js]: Burnable

File: test/Token/ERC20/ZippieInternalTokenERC20.test.js [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+const ZippieInternalTokenERC20 = artifacts.require("ZippieInternalTokenERC20");
+
+contract("ZippieInternalTokenERC20", ([owner, sender, receiver, other]) => {
+
+  beforeEach(async function () {
+    this.token = await ZippieInternalTokenERC20.new("Token (XYZ)", "TOKEN-XYZ", 6, { from: owner });
+    this.token.mint(sender, new BN(1), { from: owner })
+  });
+
+  describe('ERC20 Whitelisted Transfer (WhitelistAdminRole)', function() {
```
Comment (ripern) [inline test/Token/ERC20/ZippieInternalTokenERC20.test.js:12]: These whitelisted transfer related tests seem suitable for the issue, but why were all the ones below added also (or are they perhaps copied from somewhere into this file for some reason?)?

**Review by ripern — CHANGES_REQUESTED** [review]
Seems ok, just some concerns regarding some of the tests, comments below.

File: test/Token/ERC20/ZippieInternalTokenERC20.test.js (hunk 4 above)
Comment (p4ker) [reply test/Token/ERC20/ZippieInternalTokenERC20.test.js:12]: Want to make sure that all other ERC20 functions also works as we expect, so we can detect errors if then imported OpenZeppelin contracts are changed (new versions)

**Review by ripern — APPROVED** [review]
LGTM

**Pull request merged**